liquibase-core/src/main/java/liquibase/changelog/visitor/DBDocVisitor.java

```diff
@@ -123,7 +128,15 @@ public void writeHTML(Resource rootOutputDir, ResourceAccessor resourceAccessor)
         copyFile("liquibase/dbdoc/globalnav.html", rootOutputDir);
         copyFile("liquibase/dbdoc/overview-summary.html", rootOutputDir);
 
-        DatabaseSnapshot snapshot = SnapshotGeneratorFactory.getInstance().createSnapshot(database.getDefaultSchema(), database, new SnapshotControl(database));
+        CatalogAndSchema[] computedSchemaList = schemaList;
+        if (schemaList == null) {
+            computedSchemaList = new CatalogAndSchema[]{database.getDefaultSchema()};
+        }
+
+        DatabaseSnapshot snapshot = SnapshotGeneratorFactory.getInstance().createSnapshot(computedSchemaList, database, new SnapshotControl(database));
+        if (schemaList != null && schemaList.length != 0 && !(database instanceof FirebirdDatabase)) {
```

Comment:
May I inquire as to why there is special handling for the Firebird database?

Reply:
Firebird has some strange behaviour:
* It doesn't support schemas
* But in order to the createSnapshot to work for it, it requires a schema name of "default"
* But (again) when we do " snapshot.get(Schema.class);" in the validation, it returns no schemas, causing the validation to fail.

Besides we did a lot of stuff for Firebird recently it still buggy (a lot of tests do no work for it). So I decided to just exclude him from this validation instead of disabling dbDoc tests. If one day we fix it we can remove this condition (as in many other places).

Reply:
Thanks for the explanation; your decision makes sense to me.